### PR TITLE
[BEAM-3357] Remove upper bound for protobuf

### PR DESCRIPTION
**Pull request by aaltay**
R: @chamikaramj @robertwb

### CODE REVIEW

File: sdks/python/setup.py [hunk 1]
```diff
@@ -98,13 +98,11 @@ def get_version():
     'avro>=1.8.1,<2.0.0',
     'crcmod>=1.7,<2.0',
     'dill==0.2.6',
-    # grpcio 1.8.1 requires protobuf >= 3.5
-    # TODO(BEAM-3357): Remove the upper bound.
-    'grpcio>=1.0,<=1.7.3',
+    'grpcio>=1.0,<=1.8.1',
```
Comment (chamikaramj) [inline sdks/python/setup.py]: Can we move this to a constant and refer from this file and gen_protos ?

Comment (aaltay) [reply sdks/python/setup.py]: Having this in a single place makes sense to me. Can we do it without creating a new dependency for setup.py? Maybe gen_protos.py can extract this information from setup.py.

Do you have a suggestion?

Comment (chamikaramj) [reply sdks/python/setup.py]: In the light of allowing releases up to the upper bound, why not make the range for both grpc and grpcio-tools [1.0, 2.0) ? we can add "grpc_release_range = lambda package: '1.0>=' + package + ',<2.0'" to setup.py and use that from both setup.py and gen_protos.py.

Comment (aaltay) [reply sdks/python/setup.py]: Moved the upper bound to <2. That makes sense.

Filed https://issues.apache.org/jira/browse/BEAM-3375 for de-duplicating this dependency. (We might be fine for a while, assuming that grpcio would not come up with a major version in the near future.)

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Pull request merged**